### PR TITLE
added number of runs for optimizer

### PR DESCRIPTION
**Pull request by LianaHus**

### CODE REVIEW

File: apps/remix-ide/src/app/tabs/compileTab/compilerContainer.js [hunk 1]
```diff
@@ -182,14 +182,28 @@ class CompilerContainer {
...
+    this._view.hideWarningsBox = yo`<input class="${css.autocompile} custom-control-input" onchange=${() => this.hideWarnings()} id="hideWarningsBox" type="checkbox" title="Hide warnings">`
     if (this.data.autoCompile) this._view.autoCompile.setAttribute('checked', '')
     if (this.data.hideWarnings) this._view.hideWarningsBox.setAttribute('checked', '')
 
-    this._view.optimize = yo`<input onchange=${this.onchangeOptimize.bind(this)} class="custom-control-input" id="optimize" type="checkbox">`
+    this._view.optimize = yo`<input onchange=${() => this.onchangeOptimize()} class="custom-control-input" id="optimize" type="checkbox">`
     if (this.compileTabLogic.optimize) this._view.optimize.setAttribute('checked', '')
 
+    this._view.runs = yo`<input
+      onkeypress="return event.charCode >= 48"
```
Comment (ioedeveloper) [inline apps/remix-ide/src/app/tabs/compileTab/compilerContainer.js:194]: Is charCode consistent across all browsers?

Comment (LianaHus) [reply apps/remix-ide/src/app/tabs/compileTab/compilerContainer.js:194]: yes, should be

File: apps/remix-ide/src/app/compiler/compiler-input.js [hunk 2]
```diff
@@ -7,7 +7,7 @@ module.exports = (sources, opts) => {
     settings: {
       optimizer: {
         enabled: opts.optimize === true || opts.optimize === 1,
-        runs: 200
+        runs: opts.runs
```
Comment (yann300) [inline apps/remix-ide/src/app/compiler/compiler-input.js:10]: please use a default value (like 200) if `opts.run` is not set.

File: apps/remix-ide/src/app/tabs/compileTab/compilerContainer.js (hunk 1 above)
Comment (yann300) [inline apps/remix-ide/src/app/tabs/compileTab/compilerContainer.js:194]: what does this do? `return event.charCode >= 48`

**Comment by yann300** [comment]
you should not need any package-lock.json change in this PR

File: apps/remix-ide/src/app/tabs/compileTab/compilerContainer.js (hunk 1 above)
Comment (LianaHus) [reply apps/remix-ide/src/app/tabs/compileTab/compilerContainer.js:194]: we can remove this

**Comment by LianaHus** [comment]
> you should not need any package-lock.json change in this PR

yes I'm removeing it

File: apps/remix-ide/src/app/tabs/compileTab/compilerContainer.js [hunk 3]
```diff
@@ -174,25 +174,35 @@ class CompilerContainer {
 
   render () {
     this.compileTabLogic.compiler.event.register('compilerLoaded', (version) => this.setVersionText(version))
-    this.fetchAllVersion((allversions, selectedVersion, isURL) => {
+    this.fetchAllVersion((allversions, selectedVersion) => {
```
Comment (yann300) [inline apps/remix-ide/src/app/tabs/compileTab/compilerContainer.js:177]: `isURL` ? I think this shouldn't be part of this PR. This was added during a previous PR by @Aniket-Engg

File: apps/remix-ide/src/app/compiler/compiler-helpers.js [hunk 4]
```diff
@@ -4,17 +4,19 @@ import { Compiler } from '@remix-project/remix-solidity'
 import CompilerAbstract from './compiler-abstract'
 
 export const compile = async (compilationTargets, settings) => {
-  return await (() => {
+  let res = await (() => {
```
Comment (Aniket-Engg) [inline apps/remix-ide/src/app/compiler/compiler-helpers.js:7]: `res` can be `const`

File: apps/remix-ide/src/app/tabs/compile-tab.js [hunk 5]
```diff
@@ -213,15 +213,16 @@ class CompileTab extends ViewPlugin {
...
    * compile using @arg compilationTargets and @arg settings
    * The module UI will *not* be updated, the compilation result is returned
    * This function is used by remix-plugin compiler API.
    * @param {object} map of source files.
-   * @param {object} settings {evmVersion, optimize, version, language}
+   * @param {object} settings {evmVersion, optimize, runs, version, language}
    */
   async compileWithParameters (compilationTargets, settings) {
-    return await compile(compilationTargets, settings)
+    let res = await compile(compilationTargets, settings)
```
Comment (Aniket-Engg) [inline apps/remix-ide/src/app/tabs/compile-tab.js:224]: `const` here too

File: libs/remix-solidity/src/compiler/compiler.ts [hunk 6]
```diff
@@ -52,8 +52,9 @@ export class Compiler {
    * @param value value of key in CompilerState
    */
 
-  set <K extends keyof CompilerState>(key: K, value: CompilerState[K]): void {
+  set <K extends keyof CompilerState>(key: K, value: CompilerState[K]): void {
```
Comment (Aniket-Engg) [inline libs/remix-solidity/src/compiler/compiler.ts:55]: extra space

**Review by Aniket-Engg — COMMENTED** [review]
runs
Text box shows double carrots

File: apps/remix-ide/src/app/compiler/compiler-helpers.js (hunk 4 above)
Comment (LianaHus) [reply apps/remix-ide/src/app/compiler/compiler-helpers.js:7]: i think this was causing issues this is why i change it to let. will try const again

File: apps/remix-ide/src/app/files/file-explorer.js [hunk 7]
```diff
@@ -516,7 +516,7 @@ fileExplorer.prototype.toGist = function (id) {
         )
       } else {
         const description = 'Created using remix-ide: Realtime Ethereum Contract Compiler and Runtime. \n Load this file by pasting this gists URL or ID at https://remix.ethereum.org/#version=' +
-          queryParams.get().version + '&optimize=' + queryParams.get().optimize + '&gist='
+          queryParams.get().version + '&optimize=' + queryParams.get().optimize + '&runs' + queryParams.get().runs + '&gist='
```
Comment (Aniket-Engg) [inline apps/remix-ide/src/app/files/file-explorer.js:519]: I think it should be `runs=`

**Review by Aniket-Engg — APPROVED** [review]

File: apps/remix-ide/src/app/files/file-explorer.js (hunk 7 above)
Comment (LianaHus) [reply apps/remix-ide/src/app/files/file-explorer.js]: fixed
